src/libraries/Microsoft.PowerFx.Interpreter/Values/QueryableTableValue.cs

```diff
@@ -0,0 +1,37 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Microsoft.PowerFx.Core.IR;
+
+namespace Microsoft.PowerFx.Types
+{
+    public abstract class QueryableTableValue : TableValue
+    {
+        private readonly Lazy<Task<List<DValue<RecordValue>>>> _lazyTaskRows;
+
+        internal QueryableTableValue(IRContext irContext)
+            : base(irContext)
+        {
+            _lazyTaskRows = new Lazy<Task<List<DValue<RecordValue>>>>(() => GetRowsAsync());
+        }
+
+        internal abstract TableValue Filter(LambdaFormulaValue lambda, EvalVisitor runner, EvalVisitorContext context);
```

Comment:
>TableValue [](http://example.com/codeflow?start=26&length=10)

@lesaltzm  / @anderson-joyle  - I'm wondering if eventually all of these get pushed down to be virtuals on TableValue directly, or if it makes sense to stay here.